### PR TITLE
added tests/test_calc_barycentric_corr.py

### PR DESCRIPTION
**Pull request by cchris28**
as built verification of code against a few scenarios.
I will file an issue to verify if _calc_barycentric_corr as written is sensible as I do not understand it completely
and am not yet convinced that it does what we want it to do.

### CODE REVIEW

File: tests/test_calc_barycentric_corr.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            "im_filepath": im_filepath,
+            "res_filepath": None,
+            "time_file": None,
+            "output_suffix": "DEMO",
+            "v_arr": v_arr,
+            "ang_arr": ang_arr,
+            "bary_dist": 50.0,
+        }
+
+    def _expected_string(self, baryCoeff, shape_coeff):
```
Comment (jeremykubica) [inline tests/test_calc_barycentric_corr.py:44]: As noted below, I don't think this function is needed anymore.

File: tests/test_calc_barycentric_corr.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            for j in range(1, shape_coeff[1]):
+                value_string += f", {baryCoeff[i,j]:23.16f}"
+            value_string += f"],"
+        value_string += f"\n]\n# fmt: on"
+        return value_string
+
+    def _exception_strings(self, baryCoeff, baryExpected, shape_coeff):
+        """Return a list of strings of the actual results that violate the expected results and a boolean that is True if all the results are as expected."""
+        exception_strings = []
+        check_ok = True
```
Comment (jeremykubica) [inline tests/test_calc_barycentric_corr.py:59]: Technically you don't need `check_ok` here or below. If `exception_strings` is non-empty, then things are not okay.

File: tests/test_calc_barycentric_corr.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        # Check the barycentric correction values
+        shape_coeff = baryCoeff.shape
+        value_string = self._expected_string(baryCoeff, shape_coeff)
+        # print(value_string)
+        shape_expected = baryExpected.shape
+        assert shape_coeff == shape_expected
+        assert len(shape_coeff) == 2
+        exception_strings, check_ok = self._exception_strings(baryCoeff, baryExpected, shape_coeff)
+        if len(exception_strings) > 0:
+            print(f"Exception strings = {exception_strings}")
```
Comment (jeremykubica) [inline tests/test_calc_barycentric_corr.py:80]: Could could return `False` here and then return `True` if the if did not trigger.

File: tests/test_calc_barycentric_corr.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            [  -203.5233773925634750,     -0.0000707616325463,      0.0000001649572917,   -510.8750476689951370,      0.0000000748362879,     -0.0000703773895892],
+            [  -301.8886731240853010,     -0.0001138896682213,      0.0000002446735039,   -758.0578046608508203,      0.0000001110405992,     -0.0001133195204643],
+        ])
+        # fmt: on
+        assert self._check_barycentric(baryCoeff, baryExpected)
+
+    def test_single_image(self):
+        """Verifies that the barycentric corrections for a single image are zeros."""
+        run_search = kbmod.run_search.run_search(self.input_parameters)
+        assert run_search is not None
```
Comment (jeremykubica) [inline tests/test_calc_barycentric_corr.py:173]: `self.assertIsNotNone(run_search)`

File: tests/test_calc_barycentric_corr.py [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+            [  -102.1013035398333670,     -0.0000324007246175,      0.0000000827567619,   -256.1955780511078160,      0.0000000375305764,     -0.0000322080313739],
+            [  -102.8886591105074473,     -0.0000326747631053,      0.0000000833949229,   -258.1719728533235525,      0.0000000378200863,     -0.0000324805833438],
+            [  -201.5607485205129024,     -0.0000699606225079,      0.0000001633666874,   -505.9449406539421261,      0.0000000741141381,     -0.0000695800875374],
+            [  -202.7383401704241805,     -0.0000704409557553,      0.0000001643210668,   -508.9030324275871067,      0.0000000745474367,     -0.0000700581959558],
+            [  -203.5233773925634750,     -0.0000707616325463,      0.0000001649572917,   -510.8750476689951370,      0.0000000748362879,     -0.0000703773895892],
+            [  -301.8886731240853010,     -0.0001138896682213,      0.0000002446735039,   -758.0578046608508203,      0.0000001110405992,     -0.0001133195204643],
+        ])
+        # fmt: on
+        assert self._check_barycentric(baryCoeff, baryExpected)
```
Comment (jeremykubica) [inline tests/test_calc_barycentric_corr.py:168]: `self.assertTrue(self._check_barycentric(baryCoeff, baryExpected))`

File: tests/test_calc_barycentric_corr.py [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+        header_info.populate_from_fits_file(full_file_path)
+        time_obj = header_info.get_epoch(none_if_unset=True)
+        time_stamp = time_obj.mjd
+        visit_times = [time_stamp]
+        img_info = kbmod.image_info.ImageInfoSet()
+        img_info.append(header_info)
+        img_info.set_times_mjd(np.array(visit_times))
+        baryCoeff = run_search._calc_barycentric_corr(img_info, 50.0)
+        assert baryCoeff is not None
+        assert baryCoeff.shape == (1, 6)
```
Comment (jeremykubica) [inline tests/test_calc_barycentric_corr.py:188]: These asserts can be changed into unittest style as above.

File: setup.cfg [hunk 7]
```diff
@@ -45,6 +45,7 @@ install_requires =
     scipy>=1.9.2
     scikit_learn>=1.0.0
     koffi>=0.1.1
+    jplephem
```
Comment (jeremykubica) [inline setup.cfg:48]: Is this used anywhere?

File: tests/test_calc_barycentric_corr.py [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+import kbmod.analysis_utils
+import kbmod.search
+
+
+class test_calc_barycentric_corr(unittest.TestCase):
+    def setUp(self):
+        # Define the path for the data.
+        # TODO: this should be ../data/demo but vscode does not get the message all the time.
+        # from time import sleep
+        # sleep(300)
```
[the remark this answers is not in the document]
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:20]: yeah. missed that. Thanks.

File: tests/test_calc_barycentric_corr.py [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+        im_filepath = "../data/demo"
+        # HACK: workaround for vscode not honoring the cwd specification.
```
[the remark this answers is not in the document]
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:22]: Sort of. I think it is a bug in vscode. Interestingly I got the "run test" to work in vscode but the "debug test" just would not change to the tests directory before running. I searched and did not find a resolution so I just hacked it.
I will remove that hack.

File: tests/test_calc_barycentric_corr.py [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+
+    def _check_barycentric(self, baryCoeff, baryExpected):
+        """Check the barycentric correction values against the expected values."""
+        # Check the barycentric correction values
+        shape_coeff = baryCoeff.shape
+        value_string = self._expected_string(baryCoeff, shape_coeff)
+        # print(value_string)
+        shape_expected = baryExpected.shape
+        assert shape_coeff == shape_expected
+        assert len(shape_coeff) == 2
```
[the remark this answers is not in the document]
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:77]: ok. Is that true across the board? Should I replace all
assert foo == bar
with self.assertEqual(foo, bar)
?

File: tests/test_calc_barycentric_corr.py (hunk 2 above)
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:59]: i will change

File: tests/test_calc_barycentric_corr.py (hunk 3 above)
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:80]: will change

File: tests/test_calc_barycentric_corr.py [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+        exception_strings, check_ok = self._exception_strings(baryCoeff, baryExpected, shape_coeff)
+        if len(exception_strings) > 0:
+            print(f"Exception strings = {exception_strings}")
+        return check_ok
+
+    def _construct_wcs(
```
[the remark this answers is not in the document]
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:83]: noted

File: tests/test_calc_barycentric_corr.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (jeremykubica) [reply tests/test_calc_barycentric_corr.py:77]: That is at least the style that kbmod is using. So yes. We should replace them everywhere.

File: tests/test_calc_barycentric_corr.py (hunk 4 above)
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:173]: done

File: tests/test_calc_barycentric_corr.py (hunk 5 above)
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:168]: done

File: tests/test_calc_barycentric_corr.py (hunk 6 above)
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:188]: yup. will do

File: setup.cfg (hunk 7 above)
Comment (cchris28) [reply setup.cfg:48]: yes. It is used in _calc_barycentric_corr when a distance is specified. Which it never was until these tests.
line 383
            obs_pos = get_body_barycentric("earth", Time(mjdlist[0], format="mjd"))

File: tests/test_calc_barycentric_corr.py [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+                    )
+                    check_ok = False
+        return exception_strings, check_ok
+
+    def _check_barycentric(self, baryCoeff, baryExpected):
+        """Check the barycentric correction values against the expected values."""
+        # Check the barycentric correction values
+        shape_coeff = baryCoeff.shape
+        value_string = self._expected_string(baryCoeff, shape_coeff)
+        # print(value_string)
```
[the remark this answers is not in the document]
Comment (cchris28) [reply tests/test_calc_barycentric_corr.py:64]: I left it there because it was useful for the as-built and might be useful again for more tests.

**Review by jeremykubica — APPROVED** [review]

**Pull request merged**